code/app/src/main/java/com/example/sigma_blue/LoginFragment.java

```diff
@@ -0,0 +1,107 @@
+package com.example.sigma_blue;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+/**
+ * Class for login version of fragment from main login page
+ */
+public class LoginFragment extends DialogFragment {
+
+    private EditText usernameInput;
+    private EditText passwordInput;
+    private OnFragmentInteractionListener listener;
+
+    /**
+     * Listens for interaction with confirm button in fragment, contains method to return to
+     * original activity
+     */
+    public interface OnFragmentInteractionListener {
+        void onLoginPressed(boolean matches);
+    }
+
+    /**
+     * Creates instance to take in account object from LoginPageActivity so it can be checked
+     * against user input
+     */
+    LoginFragment newInstance(Account account) {
+
+        Bundle args = new Bundle();
+        args.putSerializable("account", account);

```

Comment:
If we have time, I think that we should swap to using parcelable instead of serializable. I think that this is alright enough to pass though.